source/common/stats/metric_impl.h

```diff
@@ -14,10 +16,10 @@ namespace Stats {
  */
 class MetricImpl : public virtual Metric {
 public:
-  MetricImpl(const std::string& name, std::string&& tag_extracted_name, std::vector<Tag>&& tags)
-      : name_(name), tag_extracted_name_(std::move(tag_extracted_name)), tags_(std::move(tags)) {}
+  MetricImpl(std::string&& tag_extracted_name, std::vector<Tag>&& tags)
+      : tag_extracted_name_(std::move(tag_extracted_name)), tags_(std::move(tags)) {}
 
-  const std::string& name() const override { return name_; }
+  const std::string name() const override { NOT_IMPLEMENTED_GCOVR_EXCL_LINE; }
```

Comment:
just delete this declaration; then classes derived from MetricImpl will be forced to override the PURE method from Metric.

Reply:
Sounds good.